keps/sig-autoscaling/4951-configurable-hpa-tolerance/README.md

```diff
@@ -0,0 +1,794 @@
+
+To avoid flapping, the number of replicas is left unchanged if this ratio is approximately 1, within a *tolerance* set to 10%. This proposal adds a parameter
+to HPAs allowing users to configure this tolerance.
+
+[Horizontal Pod Autoscaler]: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/
+[Algorithm details]: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details
+
+## Motivation
+
+Today the horizontal autoscaling tolerance is a cluster-wide parameter set using the [Kube Control Manager][] `--horizontal-pod-autoscaler-tolerance` parameter. It is by default set to 10%. While this value is often appropriate, it is considered too coarse grained in a number of scenario.
+
+This issue has been raised multiple times ([#116984][], [#125987][], [#62013][],
+[#aks-3068][], [#keda-1100][]), with users commenting that:
+
+1. For large deployments, a 10% tolerance translates into very significant resources (i.e. hundreds of pods).
+2. This tolerance can slow down scaling operations, hindering responsiveness in
+case of surges.
+3. Scale-ups are more a problem than scale-downs since typically pods are slower to initialize than to shut down, and since responding to load increase is typically more critical than freeing resources.
+
+Since appropriate tolerance values are workload-dependent, this KEP proposes to let users add custom tolerance values to `HorizontalPodAutoscaler` resources, overriding the existing default value when present.
+
+This solution integrates seamlessly with the existing HPA API since it already allows users to [fine-tune the autoscaler behavior](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#configurable-scaling-behavior).
+The exact API recommended here has been previously proposed in [kep-853][] (see [here](https://github.com/kubernetes/enhancements/pull/1234#discussion_r333036990)), but it was then proposed to implement it separately.
+
+[#116984]: https://github.com/kubernetes/kubernetes/issues/116984
+[#125987]: https://github.com/kubernetes/kubernetes/issues/125987
+[#62013]: https://github.com/kubernetes/kubernetes/issues/62013
+[#aks-3068]: https://github.com/Azure/AKS/issues/3068
+[#keda-1100]: https://github.com/kedacore/keda-docs/issues/1100
+[Kube Control Manager]: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/
+[kep-853]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-autoscaling/853-configurable-hpa-scale-velocity/README.md
+
+### Goals
+
+- Allow users to override the default workload autoscaling tolerance for specific HPAs.
+
+### Non-Goals
+
+- Allow to customize the cluster-wise tolerance given by Kube Control Manager `--horizontal-pod-autoscaler-tolerance` parameter.
+
+
+## Proposal
```

Comment:
Should we be explicit here that this will override the value of the `--horizontal-pod-autoscaler-tolerance` parameter?

Reply:
Done. (I've merged Patryk's suggestion.)